### ISSUE TITLE
Documentation bug in response templating

### ISSUE THREAD

**Issue by bkhouri** [tomakehurst/wiremock#975]
## Bug reports

The response template documentation has inconsistencies.

- WireMock Version: **v 2.18.0**
- Started using standalone via command line:  `java -jar wiremock-standalone.jar --port 8088 --verbose  --global-response-templating --local-response-templating`

Added following mapping in wiremock

```
{
  "request": {
    "urlPattern": "/template/.*"
  },
  "response" : {
    "status" : 200,
    "bodyFileName": "body-templated.json",
    "headers" : {
      "Cache-Control" : "no-cache, must-revalidate, no-store",
      "Connection" : "keep-alive",
      "Content-Type" : "application/json",
        "Date": "{{now timezone='GMT' format='EEE, dd MMM yyyy HH:mm:ss z'}}",
      "Server" : "WildFly/8",
      "Transfer-Encoding" : "chunked",
      "X-Powered-By" : "Undertow/1"
    },
    "transformers": ["response-template"]
  }
}
```

The body-templated.json file contents are

```
{
  "request.requestLine.baseUrl": "{{request.requestLine.baseUrl}}",
  "request.url": "{{request.url}}",
  "request.requestLine.path": "{{request.requestLine.path}}",
  "request.requestLine.path.0": "{{request.requestLine.path.0}}",
  "request.requestLine.path.1": "{{request.requestLine.path.1}}",
  "request.requestLine.path.2": "{{request.requestLine.path.2}}",
  "request.requestLine.path.[0]": "{{request.requestLine.path.[0]}}",
  "request.requestLine.path.[1]": "{{request.requestLine.path.[1]}}",
  "request.requestLine.path.[2]": "{{request.requestLine.path.[2]}}",
  "request.requestLine.query": "{{request.requestLine.query}}",
  "request.requestLine.query.token": "{{request.requestLine.query.token}}",
  "request.requestLine.query.relationship": "{{request.requestLine.query.relationship}}",
  "headers": {
    "request.headers.hops": "{{request.headers.hops}}"
  }
}
```

When I hit the `http://localhost:8080/template/foo?relationship=foodfg&token=mytoken`, I get the following response

```
{
  "request.requestLine.baseUrl": "http://localhost:8088",
  "request.url": "/template/foo?relationship&#x3D;foodfg",
  "request.requestLine.path": "/template/foo?relationship&#x3D;foodfg",
  "request.requestLine.path.0": "",
  "request.requestLine.path.1": "",
  "request.requestLine.path.2": "",
  "request.requestLine.path.[0]": "",
  "request.requestLine.path.[1]": "",
  "request.requestLine.path.[2]": "",
  "request.requestLine.query": "{relationship&#x3D;foodfg}",
  "request.requestLine.query.token": "",
  "request.requestLine.query.relationship": "foodfg",
  "headers": {
    "request.headers.hops": ""
  }
}
```

From the documentation found here http://wiremock.org/docs/response-templating/ (I also compared the documentation of v2.18.0 [here](https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Ftomakehurst%2Fwiremock%2Fblob%2F2.18.0%2Fdocs-v2%2F_docs%2Fresponse-templating.md&sa=D&sntz=1&usg=AFQjCNHyZJCJ2JVNGaY4cbLibWw2wwj2Jw)), states

>  `request.requestLine.path.[<n>]` - URL path segment (zero indexed) e.g. `request.path.[2]`

Based on this, I was expecting `request.requestLine.path.[0]` to return `template` and to `request.requestLine.path.[1]` return `foo`.  

According to @tomakehurst  in [this post](https://groups.google.com/forum/#!topic/wiremock-user/USFWN2JOcpk), 

> You can either do `request.path.[5]` or `request.requestLine.pathSegments.[5]`

This ticket has be open to update the documentation to reflect this.

**Comment by tomakehurst** [comment]
Fixed in 870ca8bd9a912bc9bb3efd95b228686da83e2809

**Comment by tomakehurst** [comment]
Fixed in 870ca8bd9a912bc9bb3efd95b228686da83e2809
